index.js

```diff
@@ -141,8 +170,21 @@ async function decodeTorrentFile (torrent) {
     announce: []
   }
 
-  result.infoHashBuffer = await hash(result.infoBuffer)
-  result.infoHash = arr2hex(result.infoHashBuffer)
+  // Auto-detect hash generation based on torrent type
+  const hasFileTree = hasV2Structure
+
+  const shouldGenerateV1 = hasV1Structure
+  const shouldGenerateV2 = isV2 || hasFileTree
+
+  if (shouldGenerateV1) {
+    result.infoHashBuffer = await hash(result.infoBuffer)
+    result.infoHash = arr2hex(result.infoHashBuffer)
+  }
+
+  if (shouldGenerateV2) {
+    result.infoHashV2Buffer = await hash(result.infoBuffer, undefined, 'sha-256')
+    result.infoHashV2 = arr2hex(result.infoHashV2Buffer)
+  }
```

Comment:
along with https://github.com/webtorrent/parse-torrent/pull/193/files#r2579332259 , this could be simplified 
```suggestion
  if (hasV1Structure) {
    result.infoHashBuffer = await hash(result.infoBuffer)
    result.infoHash = arr2hex(result.infoHashBuffer)
  }

  if (hasV2Structure) {
    result.infoHashV2Buffer = await hash(result.infoBuffer, undefined, 'sha-256')
    result.infoHashV2 = arr2hex(result.infoHashV2Buffer)
  }
```

Reply:
> Totally. Simplified this in 13c70e7.

Just like with Lines 142 to 144 comment above.